### PR TITLE
2808 - convert to es6 classes/modules, added imports, arrow functions, const/let

### PR DESCRIPTION
**Pull request by eleanor-heath**
https://github.com/adaptlearning/adapt_framework/issues/2808

### CODE REVIEW

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

File: js/textInputModel.js [hunk 1]
```diff
@@ -1,207 +1,201 @@
...
+    return _.pluck(this.get('_items'), 'userAnswer').join('[,]');
+  }
+
+  /**
+  * used by adapt-contrib-spoor to get the type of this question in the format required by the cmi.interactions.n.type data field
+  */
+  getResponseType() {
+    return 'fill-in';
+  }
+}
```
Comment (tomgreenfield) [inline js/textInputModel.js:201]:
```suggestion
```

File: js/textInputView.js [hunk 2]
```diff
@@ -1,108 +1,101 @@
...
-
-  });
-
-  return TextInputView;
-
-});
+    
+    this.model.get('_items').forEach((item, index) => {
+      this.$('.js-textinput-textbox').eq(index).val(item._answers[0]);
+    });
```
Comment (tomgreenfield) [inline js/textInputView.js:87]: Suggest removing duplication, e.g.
```js
const correctAnswers = this.model.get('_answers');

this.model.get('_items').forEach((item, index) => {
  const correctAnswer = correctAnswers ? correctAnswers[index][0] : item._answers[0];
  this.$('.js-textinput-textbox').eq(index).val(correctAnswer);
});
```

File: js/textInputModel.js [hunk 3]
```diff
@@ -1,207 +1,201 @@
...
-
-    isCorrect: function() {
-      if (this.get('_answers')) {
-        this.markGenericAnswers();
+import QuestionModel from 'core/js/models/questionModel';
+
+export default class TextInputModel extends QuestionModel {
+
+  init() {
+    this.genericAnswerIndexOffset = 65536;
```
[the remark this answers is not in the document]
Comment (tomgreenfield) [reply js/textInputModel.js]: The `this.genericAnswerIndexOffset` references need reverting.

File: js/textInputModel.js [hunk 4]
```diff
@@ -1,207 +1,201 @@
...
+    return this.get('_items').every(({ userAnswer }) => userAnswer);
+  }
+
+  setItemUserAnswer(itemIndex, userAnswer) {
+    const item = this.get('_items')[itemIndex];
+    item.userAnswer = userAnswer;
+    this.checkCanSubmit();
+  }
+
+  //This preserve the state of the users answers for returning or showing the users answer
```
[the remark this answers is not in the document]
Comment (tomgreenfield) [reply js/textInputModel.js]: Please would you implement the suggestion above – it doesn't read right at present.

File: js/textInputModel.js [hunk 5]
```diff
@@ -1,207 +1,192 @@
...
+  }
+
+  isCorrect() {
+    if (this.get('_answers')) {
+      this.markGenericAnswers();
+    } else {
+      this.markSpecificAnswers();
+    }
+    // do we have any _isCorrect == false?
+    return this.get('_items').some(({ _isCorrect }) => !_isCorrect);
```
Comment (tomgreenfield) [inline js/textInputModel.js:83]: Apologies, the confusing comment caused a typo; suggest removing.
```suggestion
    return this.get('_items').every(({ _isCorrect }) => _isCorrect);
```

File: js/textInputModel.js [hunk 6]
```diff
@@ -1,207 +1,192 @@
...
+
+  /**
+  * used by adapt-contrib-spoor to get the type of this question in the format required by the cmi.interactions.n.type data field
+  */
+  getResponseType() {
+    return 'fill-in';
+  }
+}
+TextInputModel.genericAnswerIndexOffset = 65536;
+export default TextInputModel;
```
Comment (tomgreenfield) [inline js/textInputModel.js:192]:
```suggestion

}

TextInputModel.genericAnswerIndexOffset = 65536;

export default TextInputModel;
```

File: js/textInputModel.js (hunk 5 above)
Comment (tomgreenfield) [reply js/textInputModel.js]: Please remove the comment.

**Review by tomgreenfield — APPROVED** [review]

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**